File: go/general/1572. 矩阵对角线元素的和.go
package main

func diagonalSum(mat [][]int) int {
	l := len(mat)
	var res int
	for i := 0; i < l; i++ {
		res += mat[i][i]
		res += mat[i][l-1-i]
	}
	if l%2 == 0 {
		return res
	} else {
		return res - mat[l/2][l/2]
	}
}

File: python/数组/1572. 矩阵对角线元素的和.py
from typing import List
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        m = len(mat)
        res = 0
        i = 0
        while i < m:
            res += mat[i][i]
            res += mat[m - 1 - i][i]
            i+=1
        if m % 2 == 0:
            return res
        else:
            return res - mat[i//2][i//2]
s = Solution()
mat = [[1,2,3],
        [4,5,6],
        [7,8,9]]
print(s.diagonalSum(mat))
